internal/envoy/route.go

```diff
@@ -137,6 +149,53 @@ func PrefixMatch(prefix string) route.RouteMatch {
 	}
 }
 
+// RateLimitStage returns stage configured for route. If none is provided
+// then the default is used
+func RateLimitStage(r *dag.Route) uint32 {
```

Comment:
Can this return a *types.UInt32Value directly, rather than requiring the caller to box it?

Also, I haven't gotten to the end of the diff yet, but if RateLimitState and RateLimitActions aren't called directly outside `internal/envoy` perhaps they can be made private to the package.